enterprise-suite/templates/es-console-configmap.yaml

```diff
@@ -206,16 +212,22 @@ data:
         etag off;
       }
 
-
-      # static files
-
-      location ~ (.*)(/assets/.*) {
+      # Specific handler for assets.
+      location ~ ^(.*)(/assets/.*)$ {
         alias $base$2;
       }
 
-      location ~ (.*)(/.*) {
+      # If there is a single level path with no trailing slash, redirect to trailing slash.
+      # E.g. "http://consoleurl/monitoring" -> 301 to "http://consoleurl/monitoring/".
+      location ~ ^/([^/]+)$ {
+        return 301 $scheme://$host$request_uri/;
```

Comment:
I don't think $host or $scheme always works and recommend just using the scheme and host from the regex to be consistent with all the others.

To test this I think you want to load through https --> http and if remembering the other one is a localhost proxy.

The idea is: we want to use the URI "as-is" and not try to rebuild it from the limited point of view of our inner nginx.

Reply:
The regex doesn't know the scheme though. How do you propose figuring it out? Good point about reverse proxies, perhaps we should just do an internal rewrite here.

Reply:
`return 301 /$1/;` would work I think?

Reply:
oh if there's an nginx behavior that auto fills a relative url, might have to do the leading space config trick which we use elsewhere. nginx is chockablock full of implicit behavior, but the idea would be to return a relative path 301 that just adds the backslash to whatever it was in the first place

leading space trick to avoid auto-fill is like line 124 (look for the other 301)

`return 301 " /$1/";